### ISSUE TITLE
umask inside npm run [...]

### ISSUE THREAD

**Issue by syndr0m** [npm/npm#6845]
npm is loosing the umask inside npm run commands :
```
$> umask 0022
$> npm config get umask
18
$> cat package.json 
{
  "scripts": {
    "umask": "npm config get umask"
  }
}
$> npm run umask
 
> @ umask /home/marc/tmp/umask
> npm config get umask

1
$> npm --version
1.4.28
```

this leads to weird files/directories permissions when using npm install inside npm run [...]

**Comment by smikes** [comment]
Hah!  I think I understand the problem here.  It's that the "022" internally gets turned into 18, which then turned into a string "18" which is then parsed again, using `parseInt(val, 8)` which silently terminates after the first char (because the `8` is an invalid octal digit).

May be fixed by my `umask` PR, #7068.

**Comment by smikes** [comment]
Found and fixed ; candidate fix on #7068

**Comment by othiym23** [comment]
Fixed by #7068, which is part of `npm@2.3.0`, which is currently `npm@next` and will be promoted to the stable version of npm this Thursday, barring something terrible.
